### ISSUE TITLE
`AttributeError` no `_plugin` with custom `Reader`

### ISSUE THREAD

**Issue by sea-shunned** [bioio-devs/bioio#66]
### Describe the Bug

When trying to print/display a `BioImage` when passing a custom reader, there is no `_plugin` attribute set:

`AttributeError: 'BioImage' object has no attribute '_plugin'`

This is because `self._plugin` is not set in the `else` clause here: https://github.com/bioio-devs/bioio/blob/main/bioio/bio_image.py#L267-L279

Happy to fix it if you want!

### Expected Behavior

Should be able to print in both cases!

### Reproduction

```python
from bioio import BioImage
import bioio_tifffile

img = BioImage("my_file.tiff")
print(img) # All good

img = BioImage("my_file.tiff", reader=bioio_tifffile.Reader)
print(img) # AttributeError
```

### Environment

-   OS Version: macOS 14.6
-   bioio Version: 1.0.3

**Comment by evamaxfield** [comment]
I think we would happily submit a pull request for this one! Thanks for finding the issue!

**Comment by sea-shunned** [comment]
Apologies for the delay, but looks like this was fixed in #67, so closing!
